### PR TITLE
[da15000] further fixes for Dialog DA15000 radio

### PR DESCRIPTION
**Pull request by JakubBrachTieto**
- ACK Frame Handling improved - race condition resolved
 - anti-collision mechanism enabled in HW (Fix for 5.8.4 cert. test)
 - filtering out broken frames
 - Frame type detection improved
 -  Radio power cycling improved
 - "Alive" blink functionality added

### CODE REVIEW

**Comment by jwhui** [comment]
@JakubBrachTieto, thanks for submitting these updates.  It appears that the code-style check is failing.  Please make sure to run `make pretty`.  Thanks.

File: examples/platforms/da15000/radio.c [hunk 1]
```diff
@@ -364,6 +387,8 @@ void otPlatRadioClearSrcMatchShortEntries(otInstance *aInstance)
 
     uint8_t i, j;
 
+    otLogDebgPlat(sInstance, "Clear ShortAddress entries", NULL);
```
Comment (aeliot) [inline examples/platforms/da15000/radio.c]: I dont think you need to pass in NULL, this should work fine with just the two arguments.

Comment (JakubBrachTieto) [inline examples/platforms/da15000/radio.c]: Using only two arguments will cause the following error during compilation:

```
error: ISO C99 requires at least one argument for the "..." in a variadic macro
     otLogDebgPlat(sInstance, "Clear ShortAddress entries");
```

File: examples/platforms/da15000/platform.c [hunk 2]
```diff
@@ -141,7 +144,22 @@ void ExampleProcess(otInstance *aInstance)
 
     if (thrValue == 0)
     {
-        hw_gpio_set_inactive(HW_GPIO_PORT_1, HW_GPIO_PIN_5);
+        // No specyfic role, let's generate alive blink
```
Comment (jwhui) [inline examples/platforms/da15000/platform.c]: handful of typos in this comment block, please proofread and fix.

**Pull request merged**